haystack/nodes/file_converter/pdf.py

```diff
@@ -176,24 +206,48 @@ def convert(
         document = Document(content=text, meta=meta, id_hash_keys=id_hash_keys)
         return [document]
 
+    def _check_tessdata(self):
+        if os.getenv("TESSDATA_PREFIX") is None:
+            raise EnvironmentError(
+                """
+                To enable OCR support via PDFToTextConverter, you need to install Tesseract:
+                    - Windows: choco install tesseract-ocr
+                    - Linux (Ubuntu): sudo apt-get install tesseract-ocr
+                    - Mac: brew install tesseract
+                After that, you need to set the environment variable TESSDATA_PREFIX to the path
+                of your Tesseract data directory. Typically this is:
+                    - Windows: C:\\Program Files\\Tesseract-OCR\\tessdata
+                    - Linux (Ubuntu): /usr/share/tesseract-ocr/4.00/tessdata
+                    - Mac (Intel):  /usr/local/Cellar/tesseract/5.3.0_1/share/tessdata
+                    - Mac (M1/M2): /opt/homebrew/Cellar/tesseract/5.3.0_1/share/tessdata
+                """
+            )
+
     def _get_text_parallel(self, page_mp):
-        idx, filename, parts, sort_by_position = page_mp
+        idx, filename, parts, sort_by_position, ocr, ocr_language = page_mp
 
         doc = fitz.open(filename)
 
         text = ""
         for i in parts[idx]:
             page = doc[i]
-            text += page.get_text("text", sort=sort_by_position) + "\f"
+            partial_tp = None
+            if ocr is not None:
+                partial_tp = page.get_textpage_ocr(
+                    flags=0, full=True if ocr == "full" else False, dpi=300, language=ocr_language
+                )
```

Comment:
```suggestion
                full = ocr == "full"
                partial_tp = page.get_textpage_ocr(
                    flags=0, full=full, dpi=300, language=ocr_language
                )
```

Reply:
@silvanocerza Much simpler than that expressive expression 😃